src/snowflake/snowpark/mock/_plan.py

```diff
@@ -1376,6 +1386,97 @@ def outer_join(base_df):
             res.append(len(deleted_row_idx))
 
         return [Row(*res)]
+    elif isinstance(source_plan, Pivot):
+        child_rf = execute_mock_plan(source_plan.child)
+
+        assert (
+            len(source_plan.aggregates) == 1
+        ), "Dataframe plan should fail before this if one aggregate isn't supplied."
+        agg = source_plan.aggregates[0]
+        assert (
+            len(agg.children) == 1
+        ), "Aggregate functions should take exactly one parameter."
+        agg_column = plan.session._analyzer.analyze(agg.children[0])
+
+        agg_function_name = agg.name.lower()
+        agg_functions = {
+            "avg": statistics.mean,
+            "count": len,
+            "max": max,
+            "min": min,
+            "sum": sum,
+        }
+
+        if agg_function_name not in agg_functions:
+            raise ValueError(
+                f"Unsupported pivot aggregation function {agg_function_name}."
+            )
+
+        pivot_column = plan.session._analyzer.analyze(source_plan.pivot_column)
+
+        if isinstance(source_plan.pivot_values, Iterable):
+            pivot_values = [exp.value for exp in source_plan.pivot_values]
+        elif source_plan.pivot_values is None:
+            pivot_values = []
+        else:
+            analyzer.session._conn.log_not_supported_error(
+                external_feature_name=f"Pivot values from {source_plan.pivot_values}",
+                internal_feature_name=type(source_plan).__name__,
+                raise_error=NotImplementedError,
+            )
+
+        grouping_columns = [
+            plan.session._analyzer.analyze(c) for c in source_plan.grouping_columns
+        ]
+        indices = grouping_columns or [
+            col for col in child_rf.keys() if col not in {agg_column, pivot_column}
+        ]
+
+        # Missing values are filled with a sentinel object that can later be replaced with Nones
+        sentinel = object()
+
+        # Snowflake treats an empty aggregation as None, whereas pandas treats it as 0.
+        # This requires us to wrap the aggregation function with extract logic to handle this special case.
+        def agg_function(column):
+            return (
+                agg_functions[agg_function_name](column.dropna())
+                if column.any()
+                else sentinel
+            )
+
+        default = (
+            source_plan.default_on_null.value if source_plan.default_on_null else None
+        )
+
+        # Count defaults to 0 rather than None
+        if agg_function_name == "count":
+            default = default or 0
+
+        result = child_rf.pivot_table(
+            columns=pivot_column,
+            values=agg_column,
+            aggfunc=agg_function,
+            index=indices,
+        )
+        result.reset_index(inplace=True)
+
+        # Pandas pivot can include many levels of indicies. we're only interested in the last one
```

Comment:
not blocker: curious what does "many levels of indicies" in pandas mean and what's the relation between the last one and other levels?

Reply:
This mostly only matter when there are multiple aggregations specified. You might get a list of columns like `[["sum", "A"], ["avg", "B"]]`. I originally assumed that since it was possible to do that with pandas it would be with snowflake as well, but after some testing it looks like snowflake only ever allows one aggregation per pivot so this probably isn't an issue. I'll remove